File: src/main/java/com/smart/appsa/model/enums/PosicaoLamina.java
package com.smart.appsa.model.enums;

import com.fasterxml.jackson.annotation.JsonCreator;

import jakarta.persistence.EnumeratedValue;

public enum PosicaoLamina {
    ESQUERDA(1),
    FRENTE(2),
    DIREITA(3);

    @EnumeratedValue
    int value;

    @JsonCreator
    public static PosicaoLamina fromValue(String value) {
        Integer parsed = Integer.valueOf(value);
        for (PosicaoLamina p : values()) {
            if (parsed.equals(p.value)) return p;
        }
        throw new IllegalArgumentException("PosicaoLamina inválida: " + value);
    }

    private PosicaoLamina(Integer value) {
        this.value = value;
    }

    public Integer getValue() { return value; }
}

File: frontend/planta-smart-4.0/src/domain/enums/PosicaoLamina.ts
export enum PosicaoLamina {
  Esquerda = 'esquerda',
  Frente = 'frente',
  Direita = 'direita',
}

export const PosicaoLaminaToInt: Record<PosicaoLamina, number> = {
  [PosicaoLamina.Esquerda]: 1,
  [PosicaoLamina.Frente]: 2,
  [PosicaoLamina.Direita]: 3,
};

export const PosicaoLaminaStringToEnum: Record<string, PosicaoLamina> = {
  ESQUERDA: PosicaoLamina.Esquerda,
  FRENTE: PosicaoLamina.Frente,
  DIREITA: PosicaoLamina.Direita,
};
